src/GLPIKey.php

```diff
@@ -118,6 +118,49 @@ public function keyExists()
         return file_exists($this->keyfile);
     }
 
+    /**
+     * Check if key is valid
+     *
+     * @return string[]
+     */
+    public function getKeyFileReadErrors(): array
+    {
+        $errors = [];
+        if (!file_exists($this->keyfile)) {
+            $errors[] = __s('You must create a security key, use `./bin/console security:change_key` command.');
+
+            return $errors; // return as, if file does not exist, no need to check further
+        }
+        if (false === ($key = @file_get_contents($this->keyfile))) {
+            $errors[] = __s("Unable to get security key file contents. Fix file permissions of {$this->keyfile}.");
+        }
+        if (strlen($key) !== SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_KEYBYTES) {
+            $errors[] = __s('Invalid security key file contents. Regenerate a key using `./bin/console security:change_key` command.');
```

Comment:
Is it really possible to get multiple errors here?
If the file is not readable, you can't really check the length of the key.

It seems there can only be one error at a time.

Reply:
> Is it really possible to get multiple errors here?
> If the file is not readable, you can't really check the length of the key.
> 
> It seems there can only be one error at a time.

For now (and for always propably), that's right. However, the method is then ready in case it happens. (Open/close principle).

Reply:
Alright, but there is still an issue IMO.

If the file is not readable, you can't check the length right after as it will always be false and thus add an unrelated "Invalid security key file contents" error while the real issue is "Unable to get security key file contents".